activerecord/lib/arel/nodes/unary.rb

```diff
@@ -35,6 +35,7 @@ def eql?(other)
       Not
       Offset
       On
+      OptimizerHints
```

Comment:
Should we ensure the hint is safe for a SQL comment, preventing query injection?

Could reuse the Arel Comment node https://github.com/rails/rails/pull/35617

Reply:
Optimizer Hints syntax is supported as not only a SQL comment but also a SQL syntax extension (e.g. use `OPTION` clause for SQL Server query level hints).

I'm not sure the desirable way to preventing query injection, Is just trimming/escaping `*/` enough?

Reply:
Sanitized it as an SQL comment https://github.com/rails/rails/pull/35615/files#diff-eaaebca1a5b748881c9e793f50b76fdfR807.